core/heap.c

```diff
@@ -892,10 +893,15 @@ vmm_heap_unit_exit(vm_heap_t *vmh)
     ASSERT(vmh->num_blocks * DYNAMO_OPTION(vmm_block_size) ==
            (ptr_uint_t)(vmh->end_addr - vmh->start_addr));
 
+#ifdef UNIX
+    bool free_heap = true;
```

Comment:
Sorry, my offline comment that it should be fine to free the whole vmm only applies to the start/stop API detach scenario where we're on the app stack.  For a regular process exit we're on the dstack, which is inside the vmm, and then we swap to the initstack, also inside the vmm: so it seems that this would crash there.

Reply:
I've updated the code and verified that it doesn't crash when running code under drrun nor when doing detach/reattach. ptal